<?xml version="1.0"?>
<launch>
  <arg name="uri"     default="radio://0/80/2M" />
  <arg name="frame"   default="rigid_body_1" />
  <arg name="joy_dev" default="/dev/input/js0" />

  <arg name="amp"     default="0.5"/>
  <arg name="freq"    default="0.5"/>
  <arg name="x_start" default="0.5"/>
  <arg name="y_start" default="0.5"/>
  <arg name="z_hover" default="0.5"/>

  <!-- Launch crazyflie server -->
  <include file="$(find crazyflie_driver)/launch/crazyflie_server.launch">
  </include>

  <group ns="crazyflie">
    <!-- Add target crazyflie to server -->
    <include file="$(find crazyflie_driver)/launch/crazyflie_add.launch">
      <arg name="uri"            value="$(arg uri)" />
      <arg name="tf_prefix"      value="crazyflie" />
      <arg name="enable_logging" value="False" />
    </include>

    <!-- Run xbox 360 controller -->
    <node name="joy" pkg="joy" type="joy_node" output="screen">
      <param name="dev" value="$(arg joy_dev)" />
    </node>

    <!-- Start controller state interpreter -->
    <node name="joystick_controller" pkg="crazyflie_demo" type="controller.py" output="screen">
      <param name="use_crazyflie_controller" value="True" />
    </node>

    <!-- Launch position PID controller -->
    <include file="$(find crazyflie_controller)/launch/crazyflie2.launch">
      <arg name="frame" value="$(arg frame)" />
    </include>

    <!-- Launch trajectory generater -->
    <node name="traj_gen" pkg="trajectory_publisher" type="trajectory_publisher" output="screen">
      <param name="amplitude" value="$(arg amp)" />
      <param name="frequency" value="$(arg freq)" />
      <param name="x_offset"  value="$(arg x_start)" />
      <param name="y_offset"  value="$(arg y_start)" />
      <param name="z_offset"  value="$(arg z_hover)" />
    </node>

    <node pkg="tf" type="static_transform_publisher" name="baselink_broadcaster" args="0 0 0 0 0 0 1 $(arg frame) /crazyflie/base_link 100" />
  </group>

  <!-- run optitrack bridge -->
  <include file="$(find optitrack)/launch/optitrack_pipeline.launch"/>

  <!-- run rviz -->
  <!-- <param name="robot_description" command="$(find xacro)/xacro.py $(find crazyflie_description)/urdf/crazyflie.urdf.xacro" /> -->
  <!-- <node name="rviz" pkg="rviz" type="rviz" args="-d $(find crazyflie_demo)/launch/crazyflie_pos.rviz" required="true" /> -->

</launch>
